subworkflows/local/prepare_references/main.nf

```diff
@@ -0,0 +1,55 @@
+include { SAMTOOLS_BGZIP } from '../../../modules/nf-core/samtools/bgzip'
+include { BWAMEM2_INDEX  } from '../../../modules/nf-core/bwamem2/index'
+
+workflow PREPARE_REFERENCES {
```

Comment:
this seems like a nice logic, having just one path for the reference will simplify things.